pkg/controllers/work/apply_controller.go

```diff
@@ -465,16 +465,13 @@ func trackDeploymentAvailability(curObj *unstructured.Unstructured) (ApplyAction
 	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(curObj.Object, &deployment); err != nil {
 		return errorApplyAction, controller.NewUnexpectedBehaviorError(err)
 	}
-	// see if the DeploymentAvailable condition is true
-	var depCond *appv1.DeploymentCondition
-	for i := range deployment.Status.Conditions {
-		if deployment.Status.Conditions[i].Type == appv1.DeploymentAvailable {
-			depCond = &deployment.Status.Conditions[i]
-			break
-		}
+	requiredReplicas := int32(1)
```

Comment:
here we're tracking the workload availability, not complete. Should we use the available condition instead?

Actually, it looks like our deamonSet and statefulset availability check is too tight. Should we respect the MaxUnavailable?

Reply:
Discussed offline, Need to update the comment in a separate PR to reflect this check.